sympy/tensor/array/expressions/array_expressions.py

```diff
@@ -288,7 +288,7 @@ def _flatten_args(cls, args):
         return new_args
 
     def as_explicit(self):
-        return Add.fromiter([arg.as_explicit() for arg in self.args])
```

Comment:
why didn't this work?

Reply:
It's creating an `Add` object instead of an array object in case `arg.as_explicit()` returns an array. It works in case it returns a matrix. I didn't investigate that much.

Do you think this should be fixed at the `Add` level?

Reply:
OK, I had a look at the code. Matrices are handled in `Add`'s postprocessors. There is no postprocessing for N-dim arrays, that's the reason it won't work. I would like to keep this PR this way as I would prefer not to venture into some edits of `Add`'s postprocessors.

Reply:
Is this not handled for matrix using kinds now? Couldn't the same be done for Array?

Reply:
> Is this not handled for matrix using kinds now? Couldn't the same be done for Array?

Probably, but I don't really want to go into the kinds for arrays until the array expressions module is finished and ready to use.